Source/WebCore/platform/graphics/GraphicsLayer.cpp

```diff
@@ -1074,7 +1074,8 @@ void GraphicsLayer::dumpProperties(TextStream& ts, OptionSet<LayerTreeAsTextOpti
             ts << childrenStream.release();
             ts << indent << ")\n"_s;
         }
-    }
+    } else
+        client().dumpRemoteChildren(ts, options);
```

Comment:
I think this statement is confusing. Does `GraphicsLayer` have to have either local children or remote children through its client? If the answer is yes then why I do not see any similar code path in this file for remote children?

This is why I do not like using `RemoteChildren` in the name. Why don't we call the new method all the times and without mentioning `Remote` in the name?
```
if (m_children.size()) {
    ....
}

client().dumpProperties(ts, options);
```

Reply:
With site-isolation, some GraphicsLayers have an identifier that references a GraphicsLayer child that lives in a different process. The actual parent-child connection only ever happens when the corresponding layer trees get pushed to the UI process (as RemoteLayerTreeNode/CALayer objects).

For normal runtime, GraphicsLayers only ever have local children, and the cross-process layer connections don't exist from the perspective of the WebProcess.

For testing, it's somewhat helpful to have `layerTreeAsText` work as if this weren't the case, and collect remote subtrees to include in the text dump. No other code in the WebProcess deals with these remote children (aside from assigning the identifiers that allow the connection to be made later).

Happy to push those details down into the client though.